core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -557,57 +559,64 @@ private Optional<Boolean> isExistingRelationForListing(Session session, Qualifie
     }
 
     @Override
-    public List<TableColumnsMetadata> listTableColumns(Session session, QualifiedTablePrefix prefix)
+    public List<TableColumnsMetadata> listTableColumns(Session session, QualifiedTablePrefix prefix, UnaryOperator<Set<SchemaTableName>> tablesFilter)
     {
         requireNonNull(prefix, "prefix is null");
 
         Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, prefix.getCatalogName());
 
         // Track column metadata for every object name to resolve ties between table and view
-        Map<SchemaTableName, Optional<List<ColumnMetadata>>> tableColumns = new HashMap<>();
+        List<Map<SchemaTableName, Optional<List<ColumnMetadata>>>> tableColumnMaps = new ArrayList<>();
```

Comment:
Did you mean `Multimap` here? Also `List` sounds like an easy object to be merged with other `List` (maybe `Set` is even better here).